Fix issue-53 by increasing socket filter buffer from 14 bytes to 16 bytes

agrath opened the pull request.

rzr commented:
Please rebase on master branch , i have bypassed failed CI job

rzr commented on lib/hci-socket/hci.js:
```diff
@@ -147,7 +147,7 @@ class Hci extends EventEmitter {
 
   setSocketFilter() {
-    const filter = Buffer.alloc(14);
+    const filter = Buffer.alloc(16);
```
Can this value computed for existing constants ?

agrath replied:
@rzr I'm sorry I don't understand what you mean here. 

This is a buffer length change. We could define a constant for the buffer length if that's what you meant.

We can see just below this line:

```
    const typeMask = (1 << HCI_EVENT_PKT) | (1 << HCI_ACLDATA_PKT);
    const eventMask1 = (1 << EVT_DISCONN_COMPLETE) | (1 << EVT_ENCRYPT_CHANGE) | (1 << EVT_CMD_COMPLETE) | (1 << EVT_CMD_STATUS) | (1 << EVT_NUMBER_OF_COMPLETED_PACKETS);
    const eventMask2 = (1 << (EVT_LE_META_EVENT - 32));
    const opcode = 0;

    filter.writeUInt32LE(typeMask, 0);
    filter.writeUInt32LE(eventMask1, 4);
    filter.writeUInt32LE(eventMask2, 8);
    filter.writeUInt16LE(opcode, 12);
```

So the value written to the buffer is constant & deterministic, and also only totals 14 bytes (3 x 32 bit = 12 bytes + 1 x 16 bit, a further 2 bytes = 14 bytes) _however_, the operating system rejects the 14 character buffer due to a kernel change (see [link](https://github.com/abandonware/node-bluetooth-hci-socket/issues/60#issuecomment-2197545245) in the issue related to this PR) as it's not aligned.

rzr replied:
ok thanks for explaining may you rebase and squash to a signgle commit with updated commit message (that explain what you just wrote here)

rzr commented on lib/hci-socket/gatt.js:
```diff
@@ -218,6 +218,15 @@ class Gatt  extends EventEmitter {
+          //above, if the characteristic defines notify, the 2902 characteristic descriptor is added automatically
+          //if the peripheral also declares this, we don't want to add it again
+          //flags (properties, secure etc) above are better defined
+          if(descriptor.uuid == '2902')
```
is there a constant name to be defined for 2902 ?

agrath replied:
There was an existing constant `GATT_CLIENT_CHARAC_CFG_UUID` defined further up at line 41, but this is declared as hex 0x2902 and we need it as a string for both the new change and the descriptor being added slightly further up.

I have changed this to declare a new constant `GATT_CLIENT_CHARAC_CFG_UUID_S` and used it in both places. Is that what you meant?

agrath commented:
> Please rebase on master branch , i have bypassed failed CI job

I think I have done that right? 
I added a remote origin "upstream", rebased issue-53 branch using upstream/master, then force pushed issue-53

> is there a constant name to be defined for 2902?

That change actually belongs to my other PR which has already been accepted and merged (but wasn't rebased). I've addressed your comment in this PR as a second commit `dfbe41d`

> Can this value computed for existing constants ?

I'm sorry, I am not sure what you mean by this. I will add more detail in the specific conversation for that line
